### PR TITLE
Date bug

### PR DESCRIPTION
**Pull request by inveterateliterate**
Setting `date ||= time` in every conversion method was resulting in always returning the date elements of the timestamp as it came in. This results in incorrect time conversions when crossing midnight.

For example: 
`utc_time = DateTime.new(2020, 3, 7, 4, 10, 20)`
`convert_timezone(utc_time, chicago_timezone, utc_timezone) >> Sat, 07 Mar 2020 23:20:00 -0500`
when it should really be `Fri, 06 Mar 2020 23:20:00 -0500`

This PR aims to resolve that.

### CODE REVIEW

File: lib/rails_util/timezone_helper.rb [hunk 1]
```diff
@@ -14,42 +14,40 @@ class << self
       # @param [Symbol=>[Date]] options key-value option pairs, used to provide a separate Date object if a specific date is needed, otherwise the date of the time argument is used
       # @return DateTime object with the time in the desired `to_timezone`, and a date of either the provided date or time
       def convert_timezone(time, to_timezone, from_timezone, **options)
-        date = options.fetch(:date, time)
+        date = options[:date]
```
Comment (bhennes2) [inline lib/rails_util/timezone_helper.rb:17]: To provide a more helpful exception message, should be keep `options.fetch` but remove the default option of `time`?

Comment (bhennes2) [reply lib/rails_util/timezone_helper.rb:17]: I'm a bit turned around on the logic here. Could the code read:
```
offset_in_seconds = Time.zone.now.in_time_zone(to_timezone).utc_offset - Time.zone.now.in_time_zone(from_timezone).utc_offset
new_datetime = DateTime.new(date.year, date.month, date.day, time.hour, time.min, time.sec, format_offset(to_timezone)) + offset_in_seconds.seconds
```

Comment (bhennes2) [reply lib/rails_util/timezone_helper.rb:17]: Perhaps we use `Time.now.in_time_zone...` instead [reference](https://stackoverflow.com/a/25143932).

Comment (inveterateliterate) [reply lib/rails_util/timezone_helper.rb:17]: I think we want the time provided! (`time.in_time_zone(timezone)`)

File: lib/rails_util/timezone_helper.rb [hunk 2]
```diff
@@ -14,42 +15,29 @@ class << self
       # @param [Symbol=>[Date]] options key-value option pairs, used to provide a separate Date object if a specific date is needed, otherwise the date of the time argument is used
       # @return DateTime object with the time in the desired `to_timezone`, and a date of either the provided date or time
       def convert_timezone(time, to_timezone, from_timezone, **options)
-        date = options.fetch(:date, time)
-        utc_time = convert_to_utc(time, from_timezone, date)
-        convert_from_utc(utc_time, to_timezone, date)
+        converted_time = time + offset_in_seconds(time, to_timezone, from_timezone)
```
Comment (inveterateliterate) [inline lib/rails_util/timezone_helper.rb:18]: @bhennes2  was able to highly simplify with your suggestion!

Added a spec to check out what happens during the other half of the year

File: .ruby-version [hunk 3]
```diff
@@ -0,0 +1 @@
+2.4.4
```
Comment (bhennes2) [inline .ruby-version:1]: Do we want a specified ruby-version if this is a gem?

File: lib/rails_util/timezone_helper.rb (hunk 2 above)
Comment (bhennes2) [reply lib/rails_util/timezone_helper.rb:18]: Nice!

File: .ruby-version (hunk 3 above)
Comment (inveterateliterate) [reply .ruby-version:1]: that crossed my mind as well, but I am hesitant to do that in this PR. I'll open an issue.

**Pull request merged**